api.planx.uk/modules/pay/middleware.ts

```diff
@@ -120,6 +125,13 @@ export async function buildPaymentPayload(
     reference: req.query.sessionId as string,
     description: "New application (nominated payee)",
     return_url: req.query.returnURL as string,
+    metadata: {
+      source: "PlanX",
+      flow:
+        req.query.returnURL.toString().split("?")?.[0]?.split("/").pop() ||
+        (req.query.returnURL as string),
```

Comment:
An example `returnURL` looks like this: https://planningservices.lambeth.gov.uk/apply-for-a-lawful-development-certificate?sessionId={uuid}&email={email}

So, this parsing: 1) Gets the URL without query params, then 2) gets the last segment (will work regardless of custom subdomain or not!)

Reply:
Doing this without a query is the way to go!

I think I'd be tempted to use the URL class (`url.pathname.split("/")[0]`) just for a bit more safety here.

Reply:
Good shout, updated !